### PR TITLE
#96 added support for decimal parameters

### PR DESCRIPTION
**Pull request by Dorofeev**

### CODE REVIEW

File: network-bignum/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    commonMainImplementation(Deps.Libs.MultiPlatform.kotlinSerialization)
+    commonMainImplementation(Deps.Libs.MultiPlatform.kbignum)
+
+    commonMainImplementation(project(":network"))
+
+    androidMainImplementation(Deps.Libs.Android.appCompat)
+
+    // temporary fix of https://youtrack.jetbrains.com/issue/KT-41821
+    commonMainImplementation("io.ktor:ktor-utils:1.4.0")
+    commonMainImplementation("org.jetbrains.kotlinx:atomicfu:0.14.4")
```
Comment (Alex009) [inline network-bignum/build.gradle.kts:25]: this part not required

File: network-bignum/src/iosMain/kotlin/Dummy.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright 2021 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.icerock.moko.network.bignum
+
+// required for produce `metadata/iosMain`
+internal val sDummyVar: Int? = null
```
Comment (Alex009) [inline network-bignum/src/iosMain/kotlin/Dummy.kt:8]: this is not required here i think

File: network-generator/src/main/kotlin/dev/icerock/moko/network/KtorCodegen.kt [hunk 3]
```diff
@@ -47,10 +48,12 @@ class KtorCodegen : AbstractKotlinCodegen() {
         typeMapping["UUID"] = "kotlin.String"
         typeMapping["URI"] = "kotlin.String"
         typeMapping["object"] = "JsonObject"
+        typeMapping["decimal"] = "BigNum"
```
Comment (Alex009) [inline network-generator/src/main/kotlin/dev/icerock/moko/network/KtorCodegen.kt:51]: for "number" should be too as i  know

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: network-generator/src/main/kotlin/dev/icerock/moko/network/KtorCodegen.kt (hunk 3 above)
Comment (Dorofeev) [reply network-generator/src/main/kotlin/dev/icerock/moko/network/KtorCodegen.kt:51]: I don't know why, but it already works for number 
here example where `amount` has format `number`, and `total` - `decimal`
```
/**
 * 
 * @param amount 
 * @param total 
 */
@Serializable
public data class DecimalTest (
    @Serializable(with = BigNumSerializer::class) 
    @SerialName("amount")
    val amount: com.soywiz.kbignum.BigNum? = null,
    @Serializable(with = BigNumSerializer::class) 
    @SerialName("total")
    val total: com.soywiz.kbignum.BigNum? = null

) 
```

**Review by Alex009 — APPROVED** [review]

**Pull request merged**